plugins/inputs/nsdp/nsdp_test.go

```diff
@@ -0,0 +1,73 @@
+package nsdp
+
+import (
+	"testing"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/parsers/influx"
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/stretchr/testify/require"
+	"github.com/tdrn-org/go-nsdp"
+)
+
+func TestInitDefaults(t *testing.T) {
+	plugin := defaultNSDP()
+	err := plugin.Init()
+	require.NoError(t, err)
+	require.Equal(t, nsdp.IPv4BroadcastTarget, plugin.Target)
+	require.Equal(t, uint(0), plugin.DeviceLimit)
+	require.Equal(t, defaultTimeout, plugin.Timeout)
+}
+
+func TestConfig(t *testing.T) {
+	conf := config.NewConfig()
+	err := conf.LoadConfig("testdata/conf/nsdp.conf")
+	require.NoError(t, err)
+	require.Len(t, conf.Inputs, 1)
+	plugin, ok := conf.Inputs[0].Input.(*NSDP)
+	require.True(t, ok)
+	err = plugin.Init()
+	require.NoError(t, err)
+	require.Equal(t, pluginTestResponderTarget, plugin.Target)
+	require.Equal(t, uint(1), plugin.DeviceLimit)
+	require.Equal(t, config.Duration(5*time.Second), plugin.Timeout)
+}
+
+const pluginTestResponderTarget = "127.0.0.1:63322"
```

Comment:
Please use the value directly in the tests instead of defining the constant. Furthemore, you cannot use a fixed port as this will cause CI to fail randomly if multiple tests are executed on the same machine and one by-chance uses this port. Use `127.0.0.1:0` to allow the OS to choose a random port...

Reply:
Adapted via 8c321f127f426a5c1cbdc9876b56cf4409fafce2